tidy-clustering/README.Rmd

```diff
@@ -0,0 +1,222 @@
+
+    +  A cluster structure being stable doesn't necessarily mean it is meaningful.
+    See e.g. [this tweet](https://twitter.com/AndrewZalesky/status/1363741266761027585)
+    
+    
+3.  **Enrichment in known labels:** (See *semi-supervised learning* above)
+
+4.  **Statistical significance/generative models:** If our clustering method  places model assumptions on how the data was generated, we can formulate a notion of statistically signficant clusters.  
+
+    +  A good example of this is "model-based clustering", which typically assumes
+    the data is generated from a mixture of multivariate Gaussians.  We can then
+    estimate the probability that a particular object came from a particular 
+    distribution in the mixture.
+    
+    
+## Other details
+
+#### Partitions versus Extractions
+
+In many clustering algorithms, each object is placed into **exactly one** cluster. This is called a **partition**.
+
+However, some algorithms allow for overlapping clusters, i.e. objects belonging to  multiple clusters - or for some objects to be "background" and have no cluster  membership at all.
+
+(This is sometimes called **community detection** or **cluster extraction**)
+
+#### What are the variables, what are the samples?
+
+Consider the example of data consisting of many gene expression measurements for several individual subjects.
+
+From a statistical perspective, we'd typically regard the subjects as samples from a population, and the gene expression levels as variables being studied.
+
+When we cluster this data, sometimes we look for clusters of **subjects**, i.e., people whose gene signature is similar.  Sometimes we look for clusters of **genes**, i.e., genes that activate or deactivate together.
+
+Non-statistical algorithms (e.g. kmeans) are agnostic to what set of objects we  cluster on.  In statistical algorithms, your model assumptions need to match your notion of samples/variables.
+
+#### Beyond the similarity matrix
+
+In the vast majority of cases, all you need for a clustering algorithm is a  *similarity matrix*; that is, entry $(i,j)$ of the matrix is the pairwise similarity  measure between object $i$ and object $j$.
+
+Of course, the choice of similarity measure matters enormously.  But the point is you only need pairwise similarities to run e.g. kmeans.
+
+Some approaches, though, need more information.  Statistical approaches in particular often estimate nuisance variables (like variance) from multiple measurements before collapsing the data into pairwise similarities.
+
+
+# Inclusion in tidymodels
+
+Here's an overview of how I see unsupervised learning fitting into a tidymodels framework.
+
+## Recipes and Workflow
+
+These steps can remain the same; there's no reason unsupervised methods need different preprocessing options/structure.
+
+## `parsnip`-like models
+
+Any and all clustering methods could be implemented as model spec functions, and the setup would look similar, e.g.
```

Comment:
I agree that a parsnip-like interface would be a great idea. I don't think that the same exact object structure and class are needed though. For example, the mode of "partition" is only required if yo fit it into an object of class `model_spec`. 

I think that an alternate class name (`cluster_spec`?) would be good with some minor difference. For example, the distance component is one that doesn't fit with a model spec.  

In other words, the api is good but using the entirety of the existing model specification format will be ungainly or kludgy.